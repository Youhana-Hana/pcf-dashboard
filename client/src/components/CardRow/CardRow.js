import React, { Component } from 'react';

import {
    Row,
    Col
} from "reactstrap";


class CardRow extends Component {
    render() {

        let value = this.props.row.value;
        let url = this.props.row.url;

      return (
              <Row className="padding-10">
              <Col xs="12" sm="6" lg="7">
              <h3>{this.props.row.label}</h3>
              </Col>
              <Col xs="12" sm="6" lg="5">
              <h3>
              { url ? <a href={url} className="text-white nav-link float-right" target="_blank">{value}</a> : value }
              </h3>
              </Col>
              </Row>
    )
  }
}

export default CardRow;
