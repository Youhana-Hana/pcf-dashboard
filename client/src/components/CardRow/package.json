{
  "name": "CardRow",
  "version": "0.0.0",
  "private": true,
  "main": "./CardRow.js"
}
