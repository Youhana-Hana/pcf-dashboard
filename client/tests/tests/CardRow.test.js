import React from 'react';
import renderer from 'react-test-renderer';
import CardRow  from '../../src/components/CardRow/';

const rowWithURL = {
  leabel: 'LABEL',
  value: 'VALUE',
  url: 'URL'
};

it('test to see if the row renders correctly', () => {
  const cardRow = renderer
  .create(<CardRow row={rowWithURL} />)
  .toJSON();
  expect(cardRow).toMatchSnapshot();
});


it('test CardRow without URL', () => {

  let row = Object.assign({}, rowWithURL);
  delete row.url;

  const cardRow = renderer
  .create(<CardRow row={row} />)
  .toJSON();
  expect(cardRow).toMatchSnapshot();
});
