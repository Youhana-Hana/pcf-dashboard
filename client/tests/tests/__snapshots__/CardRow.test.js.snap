// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the row renders correctly 1`] = `
<div
  className="padding-10 row"
>
  <div
    className="col-12 col-sm-6 col-lg-7"
  >
    <h3 />
  </div>
  <div
    className="col-12 col-sm-6 col-lg-5"
  >
    <h3>
      <a
        className="text-white nav-link float-right"
        href="URL"
        target="_blank"
      >
        VALUE
      </a>
    </h3>
  </div>
</div>
`;


exports[`test CardRow without URL 1`] = `
<div
  className="padding-10 row"
>
  <div
    className="col-12 col-sm-6 col-lg-7"
  >
    <h3 />
  </div>
  <div
    className="col-12 col-sm-6 col-lg-5"
  >
    <h3>
      VALUE
    </h3>
  </div>
</div>
`;